src/main/java/org/launchcode/water_garden_tour/controllers/AdminController.java

```diff
@@ -187,6 +181,15 @@ public String deleteGarden(Model model, @RequestParam int gardenId) {
         if (optGarden.isPresent()) {
             Garden garden = (Garden) optGarden.get();
 
+            //use method to get logged in user
+            User tourUser = userDetailServiceImplementation.getCurrentUser();
+
+            tourGardens = tourUser.getGardens();
+
+            if (tourGardens.contains(garden)) {
```

Comment:
added code to remove garden from tour before deleting garden (fixes foreign key issue in MySQL when deleting a garden)

Reply:
I didn't catch that error.  The error doesn't happen in the master, is that a result of fixing the validation?